File: code-coaches/hard/hofstadters-q-sequence/hofstadters_q_sequence.rb
def q_sequence(n)
  return 1 if n <= 2

  q_sequence(n - q_sequence(n - 1)) + q_sequence(n - q_sequence(n - 2))
end

def main
  n = gets.chomp.to_i
  puts q_sequence(n)
end

if __FILE__ == $0
  main
end

File: code-coaches/hard/hofstadters-q-sequence/hofstadters_q_sequence.py
def q_sequence(n: int) -> int:
    if n <= 2:
        return 1
    else:
        return q_sequence(n - q_sequence(n - 1)) + q_sequence(n - q_sequence(n - 2))


def main():
    n = int(input())
    print(q_sequence(n))


if __name__ == "__main__":
    main()
